ivy/functional/ivy/experimental/manipulation.py

```diff
@@ -1782,36 +1809,54 @@ def put_along_axis(
     >>> values = ivy.array([[9, 8, 7], [6, 5, 4]])
     >>> ivy.put_along_axis(arr, indices, values, 1, mode='clip')
     >>> print(arr)
-    ivy.array([[3, 7, 5],
-               [6, 4, 1]])
+    ivy.array([[9, 7, 5],
+               [4, 2, 6]])
+
+    >>> arr = ivy.array([[10, 30, 20], [60, 40, 50]])
+    >>> axis = 1
+    >>> indices = ivy.argmax(arr, axis=axis, keepdims=True)
+    >>> value = 100
+    >>> ivy.put_along_axis(arr, indices, value, axis, mode='add')
+    >>> print(arr)
+    ivy.array([[ 10, 130, 20],
+               [ 160, 40, 50]])
     """
-    if out is None:
-        out = ivy.zeros_like(arr)
-
-    indices = ivy.expand_dims(indices, axis=axis)
-    values = ivy.expand_dims(values, axis=axis)
+    if len(arr.shape) != len(indices.shape):
+        raise ivy.utils.exceptions.IvyValueError(
+            "indices and arr must have the same number of dimensions"
+        )
+    # out = ivy.zeros_like(arr) if out is None else out
+    broadcast_shape = ivy.broadcast_shapes(*[arr.shape, indices.shape, values.shape])
+    values = ivy.asarray(values) if not isinstance(values, ivy.Array) else values
+    if broadcast_shape:
+        indices = ivy.broadcast_to(indices, broadcast_shape)
+        values = ivy.broadcast_to(values, broadcast_shape)
 
-    stacked = ivy.concat((arr, values), axis=axis)
+    Ni, Nk = arr.shape[:axis], arr.shape[axis + 1 :]
+    ret = []
 
-    sorted_indices = ivy.argsort(indices, axis=axis)
-    sorted_stacked = ivy.take_along_axis(stacked, sorted_indices, axis=axis)
+    for ii in _ndindex(Ni):
+        for kk in _ndindex(Nk):
+            arr_1d = arr[ii + (slice(None),) + kk]
+            indices_1d = indices[ii + (slice(None),) + kk]
+            values_1d = values[ii + (slice(None),) + kk]
+            a = _handle_reduction_op(mode, arr_1d, indices_1d, values_1d)
+            ret.append(a)
 
-    arr = ivy.where(
-        ivy.expand_dims(sorted_indices < arr.shape[axis], axis=axis),
-        sorted_stacked,
-        arr,
-    )
+    ret = ivy.vstack(ret).T
```

Comment:
@AnnaTz for some inputs I have to get the transpose at this stage i.e `ivy.vstack(ret).T`, but I don't know what is causing this issue. This was the main pain point in this PR. Otherwise everything else is working fine